src/profile-logic/sanitize.js

```diff
@@ -491,6 +496,42 @@ function sanitizeThreadPII(
           }
         }
       }
+
+      // Now we'll remove samples related to the frames
+      // First we'll loop the stack table and populate a typed array with a flag
+      // indicating "this stack index has an ancestor with the private browsing
+      // flag". This means children inherits the private browsing flag. This is
+      // possible because when iterating we visit parents before their children.
+      const isStackPrivateBrowsing = new Uint8Array(stackTable.length);
+      for (let stackIndex = 0; stackIndex < stackTable.length; stackIndex++) {
+        const prefix = stackTable.prefix[stackIndex];
+        if (prefix !== null && isStackPrivateBrowsing[prefix]) {
+          isStackPrivateBrowsing[stackIndex] = 1;
+          continue;
```

Comment:
For this case, we know that the `prefix` stack must always be present in the array "before" the the children stacks, right?

Reply:
yes :-)